pkg/storage/bucket/s3/bucket_client.go

```diff
@@ -7,6 +7,11 @@ import (
 	"github.com/thanos-io/objstore/providers/s3"
 )
 
+const (
+	// Applied to PUT operations to denote the desired storage class for S3 Objects
+	awsStorageClassHeader = "X-Amz-Storage-Class"
```

Comment:
is it case-insensitive? 
AWS Docs says that it must be `x-amz-storage-class` https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html

Reply:
minio lib confirms that it can be :
```golang
	// Storage class header.
	amzStorageClass = "X-Amz-Storage-Class"
```

so, ignore my comment